hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/strategy/BoundedIOCompactionStrategy.java

```diff
@@ -44,10 +44,10 @@ public List<HoodieCompactionOperation> orderAndFilter(HoodieWriteConfig writeCon
     for (HoodieCompactionOperation op : operations) {
       long opIo = op.getMetrics().get(TOTAL_IO_MB).longValue();
       targetIORemaining -= opIo;
-      finalOperations.add(op);
-      if (targetIORemaining <= 0) {
+      if (targetIORemaining < 0) {
         return finalOperations;
       }
+      finalOperations.add(op);
```

Comment:
This can lead to starvation if the target IO limit is always smaller than the `TOTAL_IO_MB` of the first compaction operation.